src/Libraries/Microsoft.Extensions.AI/ChatCompletion/FunctionInvokingChatClient.cs

```diff
@@ -82,6 +84,9 @@ public static FunctionInvocationContext? CurrentContext
         protected set => _currentContext.Value = value;
     }
 
+    /// <summary>Gets the <see cref="IServiceProvider"/> used for resolving services required by the <see cref="AIFunction"/> instances being invoked.</summary>
+    public IServiceProvider? Services { get; }
```

Comment:
I'm not certain why this is public API. Could you clarify?

Reply:
It doesn't have to be. I could make it private if you'd prefer.

Reply:
I'd default to private if we have the choice.